### ISSUE TITLE
Making Chatbot reply to personal questions

### ISSUE THREAD

**Issue by xames3** [daniel-kukiela/nmt-chatbot#107]
**Hi Daniel...** 

Short description of what I'm working with:
I'm training my bot with about **20M** rows, train.from and train.to has about **20,513151** rows. Training using Paperspace (feel free to share any and all for referral codes 😝) for about 5 days roughly on a P5000 **16 GB** VRAM machine. Epoch is for **160,259** steps. 1 Epoch is completed with learning rate: 0.001 (first learning rate). Current model checkpoint step: **187,259** steps. Model is about 50-50% coherent as of now.

**Vocabulary size :** 230,000
Rest all is unchanged (**batchsize = 128**). Not sure how big is vocab.bpe files are.

### My questions are:

1. Will it be possible for adding any placeholders like **[PLACEHOLDER]** for adding personal results?
Like I want to give my chatbot a name and sex (say: Scot and male), so when I ask, 
#### Who are you?
>>**(Ideally)**
>> Scot
 **(I'm getting this)**
>> I don't know what you're talking about/Incomplete Wikipedia link/I'm not sure what to say.
**(I want to implement this)**
>> I'm **[PLACEHOLDER_FOR_NAME]** so output will be **I'm Scot.**

#### Are you a girl or a boy?
>> Male/Boy/Man **(Ideally)**
>> Sometimes it says boy sometime it says girl **(Inconsistency in results)**

2. Will it be possible to ask the bot to do somethings on behalf of me, like I've made a script for pulling gmail unread counts... I want the chat bot to read the unread mail counts.

#### Scot how many unread mails do I have?
>>**(Ideally)**
>> 2981 mails (Pulling information from my other script. I can add it in Inference.py script, but not sure how to implement it here in this case)
 **(I'm getting this)**
>> 19 or 23 (any random number).
**(I want to implement this)**
>> You have about **[PLACEHOLDER_FOR_UNREAD_COUNT]** mails pending. 
so output will be **You've about **2981** mails pending**

### There are few other questions too but these were\are my primary concern.
Request your help here.
Thanks in advance!

**Comment by AquaDef** [comment]
I can help you a bit.
Try adding your own data into the database, How? By downloading data browser for sqlite3 then adding a row with the question give it an id the write the answer with the same id

Note: This has not been tested it's just an idea or a concept.

**Comment by xames3** [comment]
> I can help you a bit.
> Try adding your own data into the database, How? By downloading data browser for sqlite3 then adding a row with the question give it an id the write the answer with the same id
> 
> Note: This has not been tested it's just an idea or a concept.

Thanks for looking into this @AquaDef . I really appreciate your help and suggestion. Technically, I did exactly the same thing here[https://github.com/xames3/alice](url), This was my second iteration of chatbot. It worked well. Again, thanks for the suggestion!!!
